TypeSystem/flow-analysis/reachability_for_A03_t05.dart

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion for statement: If `N` is a for statement of the form
+/// `for (D; C; U) S,` then:
+/// - Let `before(D) = before(N)`.
+/// - Let `before(C) = conservativeJoin(after(D), assignedIn(N), capturedIn(N))`
+/// - Let `before(S) = split(true(C))`.
+/// - Let `before(U) = merge(after(S), continue(S))`.
+/// - Let `after(N) = inheritTested(join(false(C), unsplit(break(S))), after(U))`
+///
+/// @description Checks that `before(U) = merge(after(S), continue(S))`. Test
+/// that if the static type of `D` is `Never`, then an assignment in `U` will
+/// never happen.
```

Comment:
OK, let's just keep this one where we are testing the treatment of dead code: It has comments to remind a reader that the code is dead, and it seems pragmatic/useful for the analysis to avoid reporting any other diagnostics than 'dead code' for line 23, 31, 40.

However, I don't see any cases where `D` has type `Never`. Missing update in `@description`?

Reply:
Yes, thank you.  Description update was missing. Fixed.